### ISSUE TITLE
Handling Authorization

### ISSUE THREAD

**Issue by BjoernRave** [FormidableLabs/next-urql#3]
How would you handle authorization, for example if your authorization token is stored as a cookie since you don't have access to the Context

**Comment by parkerziegler** [comment]
Interesting question, I haven't thought too much about this. I suppose my initial thought is to use an additional HoC on top of `withUrqlClient`, for example: https://github.com/zeit/next.js/blob/canary/examples/with-cookie-auth/utils/auth.js It does get into the classic multiple wrappings of HoCs issue, but what I like about this approach is that it lives in user-land. `urql` should remain agnostic to _how_ it receives an auth token. Some pseudo-code might look like this:

```javascript
// The component you want to wrap.
const Page = () => (
   <div>Page</div>
);

const injectToken = (props) =>
   withUrqlClient({
      url: 'https://myendpoint.com/graphql',
      fetchOptions: {
         headers: { Authorization: `Bearer${props.token}` }
      }
   })(Page);

export default withAuthSync(injectToken);
```

Again, I'm not fully confident the above will work as is, it's just some pseudo-code to get at the idea. We could consider exposing the context directly on our end as well, where the first argument to `withUrqlClient` is a function that's handed context and returns the client configuartion. So the API would change to look like:

```typescript
withUrqlClient((ctx: NextContext) => ({ /* your client config options */ }))(Component)
```

But I'm not 💯 % confident in that implementation just yet. What are your thoughts?

**Comment by kylealwyn** [comment]
+1 to adding support for the generator fn. would love to do something like
```js
   withUrqlClient(ctx => ({
      url: 'https://myendpoint.com/graphql',
      fetchOptions: {
         headers: { Authorization: process.browser ? cookies.get('accessToken') : cookies.get(ctx, 'accessToken' }
      }
   }))(Page);
```

**Comment by parkerziegler** [comment]
Awesome, I'll take a look at integrating `ctx` into that API this week, but I suspect we'd be able to make that happen!

**Comment by parkerziegler** [comment]
Released in v0.2.0: https://github.com/FormidableLabs/next-urql/releases/tag/v0.2.0
